### PR TITLE
Make Git workflow a bit more clear

### PR DESCRIPTION
**Pull request by AlexCatch**
## Description

I don't think we explicitly specify when to use each branch enough which ends up with issues like branches being called `modification/*`

There are instances where people who opened a PR didn't know that we group all new features, modifications & non essential fixes into `feature` branches as to not end up with loads of different branch prefixes.

This should clear it all up.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds documentation)
- [ ] Breaking change (This is a significant change)

## Checklist

- [x] I have ensured that I have used relative links where appropriate.
- [x] I have proofread my documentation.

### CODE REVIEW

**Comment by dextermb** [comment]
While you're right about non-essential fixes going into `feature/*` branches I think a number of projects have taken up the `fix/*` naming convention.

Perhaps this is something that should be considered.

**Comment by ash123456789** [comment]
> While you're right about non-essential fixes going into `feature/*` branches I think a number of projects have taken up the `fix/*` naming convention.
> 
> Perhaps this is something that should be considered.

I had proposed something along the lines of an `improvement/*` or a `fix/*` branch. I also think that using some sort of improvement/fix branch would be ideal instead of just lumping everything under `features/*`.

https://github.com/langleyfoxall/technologies/issues/41

**Comment by AlexCatch** [comment]
I don't really see the need to distinguish between them, a feature to me is a change to the code base so encompasses all changes, that's what I do in my personal projects anyway.

As with all standards, it doesn't matter what it is as long as it's consistent. I'm not that fussed as long as we have a fixed standard.

**Comment by AlexCatch** [comment]
@MaxHarrisonGit I haven't just added it in, it's what we had discussed previously but never got into words/fleshed out. I'm sorry if it comes across as an attack because that wasn't the intention. I just assumed you were aware because our Git flow has been discussed previously in the repo.

**Comment by wheatleyjj** [comment]
+1 for `fix/*` over using `feature/*` for fixes

**Comment by DivineOmega** [comment]
I quite like the idea of using `fix/*` where appropriate but I'm not hugely concerned. Branch names are very temporary, because of how we work (deleting branches immediately after PR merge).

**Comment by DivineOmega** [comment]
@AlexCatch Would you mind resolving conflicts on this PR, when you're free and perhaps look at adding the `fix/*` as discussed?

File: devops/git/workflow.md [hunk 1]
```diff
@@ -8,8 +8,8 @@ Branches should be harmonious between projects and follow a set standard. Branch
 after the initial commit, with the correct protection rules in place.
 
 * `master` should be the working development branch [UNSTABLE]
-* `feature/*` branches created from master for each feature
-* `hotfix/*` branches should only be created off of `staging` or `production` and merged accordingly.
+* `feature/*` branches created from master for each feature, change or fix
```
Comment (dextermb) [inline devops/git/workflow.md]: I still like the idea of `fix/*` branches. Perhaps we can have a vote on this and add it in the same PR if won?

**Comment by robmilward** [comment]
I feel like this could do with a bit more of an in-depth explanation.

For instance, there's ambiguity surrounding what to do with a `hotfix` branch. Yes, we branch off of `staging` or `production`, but what does it mean to 'merge accordingly'?
Given it suggests we can merge off of staging _or_ production, perhaps it's worth highlighting the similarities and differences this flow is implementing in regards to the commonly used [Gitflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) (which isn't _that_ actually far from what we're trying to implement anyway).

image

Was there a reason Gitflow was avoided in the first place?




> @AlexCatch perhaps look at adding the `fix/*` as discussed?

I feel like the semantics of a branch name should indicate how it's being treated whilst being merged back into the code base, and what it represents.. not necessarily what the thing you're writing actually is. 

For instance: 
- `feature` branches - (standard Gitflow) always off of the development branch, merged back into the development branch
- hotfix - (standard Gitflow) always off of the production branch, back into production and development simultaneously.

But a fix branch? Would you only ever use a fix branch in place of a feature branch? Would you ever, in any case, treat it the same as a hotfix? What if the _thing_ you're implementing isn't really a fix OR a feature - would we want another type of branch for that? 


IMO fix/ adds unnecessary cognitive overhead and acts as a slippery slope to the exact point this PR was trying to avoid.

**Comment by AlexCatch** [comment]
@robmilward I've always reccomend following standard gitflow but that was voted against.

**Pull request closed without merging**